bullet_train-super_load_and_authorize_resource/app/controllers/concerns/bullet_train/loads_and_authorizes_resource.rb

```diff
@@ -94,35 +96,28 @@ def account_load_and_authorize_resource(model, positional_through = nil, through
 
       # x. this and the thing below it are only here to make a sortable concern possible.
       prepend_before_action only: member_actions do
-        instance_variable_name = options[:polymorphic] || through_as_symbols[0]
-        eval "@child_object = @#{model}"
-        eval "@parent_object = @#{instance_variable_name}"
+        @child_object = instance_variable_get("@#{model}")
+        @parent_object = instance_variable_get instance_variable_name
       end
 
       prepend_before_action only: collection_actions do
-        instance_variable_name = options[:polymorphic] || through_as_symbols[0]
-        eval "@parent_object = @#{instance_variable_name}"
-        if options[:through_association].present?
-          eval "@child_collection = :#{options[:through_association]}"
-        else
-          eval "@child_collection = :#{model.to_s.pluralize}"
-        end
+        @parent_object = instance_variable_get instance_variable_name
+        @child_collection = options[:through_association].presence&.to_sym || model.to_s.pluralize.to_sym
       end
 
       prepend_before_action only: member_actions do
-        instance_variable_name = options[:polymorphic] || through_as_symbols[0]
-        possible_sources_of_parent =
-          through_as_symbols.map { |tas| "@#{model}.#{tas}" }.join(" || ")
-        eval_string =
-          "@#{instance_variable_name} ||= " + possible_sources_of_parent
-        eval eval_string
+        if (model_instance = instance_variable_get("@#{model}")) && !instance_variable_defined?(instance_variable_name)
```

Comment:
@kaspth, was this intentionally an assignment inside of a conditional check? Or should this be `==`? Since we use `model_instance` below I'm guessing the assignment is intentional. If that's the case I think this would be nicer to read:

```suggestion
        model_instance = instance_variable_get("@#{model}")
        if model_instance && !instance_variable_defined?(instance_variable_name)
```

Reply:
It was! I just merged your suggestion in, since you're right it's clearer (I got a little confused trying to get what I was going for here at first).